pypesto/visualize/sampling.py

```diff
@@ -91,6 +114,565 @@ def sampling_fval_trace(
+                    condition_index,
+                    observable_id,
+                )
+                t_lower_shifted = t_lower + t0
+                t_upper_shifted = t_upper + t0
+                # Timepoints must match, or `upper_data` will be plotted at
+                # some incorrect time points.
+                assert (np.array(t_lower) == np.array(t_upper)).all()
+                ax.fill_between(
+                    t_lower_shifted,
+                    lower_data,
+                    upper_data,
+                    facecolor=rgba2rgb(
+                        variables_color[condition_index]
+                        + [levels_opacity[level_index]]
+                    ),
+                    lw=0,
+                )
+                t_max = max(t_max, *t_lower_shifted, *t_upper_shifted)
+            t0 = t_max
+
+
+def sampling_prediction_trajectories(
+        ensemble_prediction: EnsemblePrediction,
+        levels: Union[float, Sequence[float]],
+        title: str = None,
+        size: Tuple[float, float] = None,
+        axes: matplotlib.axes.Axes = None,
+        labels: Dict[str, str] = None,
+        axis_label_padding: int = 30,
+        groupby: str = CONDITION,
+        condition_gap: float = 0.01,
+) -> matplotlib.axes.Axes:
+    """Plot MCMC-based prediction confidence intervals for the
+    model states or observables. One or various confidence levels
+    can be depicted. Plots are grouped by condition.
+
+    Note that references to observables here corresponds to the meaning of an
+    observable in the pyPESTO predictions code, which means an observable could
+    refer to many things, including model states or observables, dependent on
+    how the prediction is processed. FIXME rename
+
+    Parameters
+    ----------
+    result:
+        The pyPESTO result object with filled sample result.
+    levels:
+        Credibility levels, e.g. [95] for a 95% credibility interval. See the
+        :py:func:`_get_level_percentiles` method for a description of how these
+        levels are handled, and current limitations.
+    title:
+        Axes title.
+    size: ndarray
+        Figure size in inches.
+    axes:
+        Axes object to use.
+    labels:
+        Keys should be ensemble observable IDs, values should be the desired
+        label for that observable. Defaults to observable IDs.
+    axis_label_padding:
+        Pixels between axis labels and plots.
+    groupby:
+        Group plots by `OBSERVABLE` or `CONDITION`. TODO single constants file?
+    condition_gap:
+        Gap between conditions when `groupby == CONDITION`.
+
+    Returns
+    -------
+    axes:
+        The plot axes.
+    """
+    if labels is None:
+        labels = {}
+    if len(list(levels)) == 1:
+        levels = list(levels)
+    levels = sorted(levels, reverse=True)
+    percentiles = [
+        percentile
+        for level in levels
+        for percentile in _get_level_percentiles(level)
+    ]
```

Comment:
Would also be happy about a line of comment here, which optically structures this a block, such as "prepare by processing percentile levels". No big thing needed, but I just like the style of having the code coming in small portions, which have a short comment that tells you what is done. If you scroll through a routine, that helps you to see whether you can skip this part of code (because it does probably not contain what you're interested in) or whether you want to take the time to read it carefully